src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -23,6 +24,18 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
     {
         private const int MaxResults = 3;
 
+        private readonly IPackageInstallerService _packageInstallerService;
```

Comment:
If they're null then we will attempt to retrieve them from the workspace.  But this approach allows us to simply replace the workspace injected versions with our own mocks at test time.